Libraries/StyleSheet/StyleSheet.js

```diff
@@ -69,4 +72,7 @@ class StyleSheet {
   }
 }
 
+/* TODO(brentvatne) docs are needed for this */
+StyleSheet.flatten = flattenStyle;
```

Comment:
I would like to move this logic to React and decouple this from styles. The refactoring I've done has decoupled this logic from the `style` prop.

This is similar to the `React.Children.toArray` helper so something like that on `React` might make sense.

It doesn't really hurt to have it here too for now but I don't think it ultimately belongs here.